genisys/main.py

```diff
@@ -13,15 +17,21 @@ def validate(modules):
 
 def install_config(file, root="/"):
     print(f"Installing config file: {file} with root at {root}")
+    # netplan
+    netplan = net.Netplan(file)
```

Comment:
It looks like when you call this function on line 68, you pass the filename as the `file` parameter. All the modules expect to receive an instance of `YAMLParser`. You're making one on line 44, it just needs passed into this function.